src/server.c

```diff
@@ -1168,6 +1168,21 @@ void cronUpdateMemoryStats() {
     }
 }
 
+/* Pause clients on write as long as OOM and pending lazy-free jobs in background.
+ * By doing so the server might avoid unnecessary eviction or rejection. */
+static void ClientWritePauseDuringOOM() {
+    
+    /* if OOM and available pending lazyfree jobs then pause client write */
+    if ((getMaxmemoryState(NULL,NULL,NULL,NULL)) &&
```

Comment:
getMaxmemoryState needs to be checked against C_OK / C_ERR (don't rely on the fact C_OK is 0)